### ISSUE TITLE
How to shorten some headlines in TOC by headline attribute in content

### ISSUE THREAD

**Issue by matteoFeduzi** [tscanlin/tocbot#247]
Some of my headlines are long and in the TOC we would like to customize them by creating shorter, more impactful text.

Is it possible to use an attribute to place on each headline you want to shorten like the following?
`data-label-for-toc`

For example in my headline I will have the following code:
`<h2 data-label-for-toc="headline">Headline with more text</h2>`

and on the TOC instead of "Headline with more text" I will see "headline".

This is really important to me.

I've tried looking through the various tickets that are already open, but I don't really understand how to do this.

Thanks!

**Comment by tscanlin** [comment]
hmm, the second one should work if you change the strings to match your use case. What is the name of the heading you are trying to change?

**Comment by tscanlin** [comment]
I put up a PR with a change that should handle your use case. Can you check it out and let me know if it helps you?
https://github.com/tscanlin/tocbot/pull/250/files

**Comment by matteoFeduzi** [comment]
This is the answer to your first comment:
Hi, I just updated the code. I had the old version and it wasn't working on purpose. The problem though is that for all the other headlines that I don't want to edit it returns "undefined" as in the screen below:
https://monosnap.com/file/c0wHCW2Nrrw0WAlG4hkqwb012BVgNw

And this is the answer to your second comment:
Unfortunately, I'm not very hands-on and don't understand what I'm supposed to do with the thread you opened. Should I perhaps make those changes that I see? Because I tried to search the code, but I can't find those lines. (Sorry, but I'm not very practical)

**Comment by tscanlin** [comment]
Hey no worries. If you try using the code here (copy / paste into a script tag) then you should be able to test it out:
https://github.com/tscanlin/tocbot/pull/250/files#diff-72f16c3781a35e8629f753301c06bd2cfb0ecb6436c1fcc126cee7e3343e3670R1

**Comment by matteoFeduzi** [comment]
Now I have this code:
```
function o(t) {
                if (!(t instanceof window.HTMLElement)) return t;
                if (e.ignoreHiddenElements && (!t.offsetHeight || !t.offsetParent)) return null;
				const headingLabel = heading.getAttribute('data-heading-label')
      || (options.headingLabelCallback ? String(options.headingLabelCallback(heading.textContent)) : heading.textContent.trim())
                var n = {
                    id: t.id,
                    children: [],
                    nodeName: t.nodeName,
                    headingLevel: (function (e) {
                        return +e.nodeName.split("H").join("");
                    })(t),
                    textContent: headingLabel,
                };
                return e.includeHtml && (n.childNodes = t.childNodes), e.headingObjectCallback ? e.headingObjectCallback(n, t) : n;
}
```
But rightly `heading` is not called correctly.

How do I call it correctly?

Thank you!

**Comment by tscanlin** [comment]
What do you mean it's not called correctly? Do you have an example page you can link me to?

**Comment by tscanlin** [comment]
here is also a prerelease link on a CDN which should work
https://cdnjs.cloudflare.com/ajax/libs/tocbot/4.13.6-1/tocbot.js

**Comment by matteoFeduzi** [comment]
Yes, sure. Here the file with your JS script in my staging website:
https://sviluppo.sos-wp.it/wp-content/themes/sos-wp/js/post.js?ver=1.1

And this is the one of the page where I want implement the TOC:
https://sviluppo.sos-wp.it/creare-siti-web/multilingua/come-creare-un-sito-multilingua/

**Comment by matteoFeduzi** [comment]
Great, now it works. Simply needed to update the code. Thanks again!

**Comment by tscanlin** [comment]
Glad it worked, thanks for helping to test it out! I appreciate your contribution :)

This change has been merged and is published in `tocbot@4.16.0`
